dex/governance/src/config.rs

```diff
@@ -32,78 +33,98 @@ pub trait Config {
         self.try_change_voting_period_in_blocks(new_value);
     }
 
+    #[endpoint(changeGovernanceTokenIds)]
+    fn change_governance_token_ids(&self, token_ids: ManagedVec<TokenIdentifier>) {
+        self.require_caller_self();
+
+        self.try_change_governance_token_ids(token_ids);
+    }
+
+    #[endpoint(changePriceProviders)]
+    fn change_price_providers(
+        &self,
+        #[var_args] price_providers: MultiValueEncoded<
+            MultiValue2<TokenIdentifier, ManagedAddress>,
+        >,
+    ) {
+        self.require_caller_self();
+
+        self.try_change_price_providers(price_providers);
+    }
+
     fn require_caller_self(&self) {
         let caller = self.blockchain().get_caller();
         let sc_address = self.blockchain().get_sc_address();
 
-        require!(
-            caller == sc_address,
-            "Only the SC itself may call this function"
-        );
+        require!(caller == sc_address, INVALID_CALLER_NOT_SELF);
     }
 
     fn try_change_mex_token_id(&self, token_id: TokenIdentifier) {
-        require!(
-            token_id.is_esdt(),
-            "Invalid ESDT token ID provided for vote_nft"
-        );
+        require!(token_id.is_esdt(), INVALID_ESDT);
 
         self.mex_token_id().set(&token_id);
     }
 
     fn try_change_vote_nft_id(&self, token_id: TokenIdentifier) {
-        require!(
-            token_id.is_esdt(),
-            "Invalid ESDT token ID provided for vote_nft"
-        );
+        require!(token_id.is_esdt(), INVALID_ESDT);
 
         self.vote_nft_id().set(&token_id);
     }
 
     fn try_change_governance_token_ids(&self, token_ids: ManagedVec<TokenIdentifier>) {
-        for token_id in token_ids.iter() {
-            require!(
-                token_id.is_esdt(),
-                "Invalid ESDT token ID provided for token_ids"
-            );
+        self.governance_token_ids().clear();
+
+        for token_id in token_ids.into_iter() {
+            require!(token_id.is_esdt(), INVALID_ESDT);
+
+            self.governance_token_ids().insert(token_id);
         }
+    }
+
+    fn try_change_price_providers(
+        &self,
+        #[var_args] price_providers: MultiValueEncoded<
+            MultiValue2<TokenIdentifier, ManagedAddress>,
+        >,
+    ) {
+        self.price_providers().clear();
 
-        self.governance_token_ids().set(&token_ids);
+        for provider in price_providers.into_iter() {
+            let tuple = provider.into_tuple();
+            require!(tuple.0.is_esdt(), INVALID_ESDT);
+            require!(!tuple.1.is_zero(), ZERO_VALUE);
+
+            self.price_providers().insert(tuple.0, tuple.1);
+        }
     }
 
     fn try_change_quorum(&self, new_value: BigUint) {
-        require!(new_value != 0u64, "Quorum can't be set to 0");
+        require!(new_value != 0u64, ZERO_VALUE);
```

Comment:
`BAD_ZERO_VALUE`, perhaps - and above as well (optionally)? (so that we know it should not be zero, without ambiguity).

Reply:
done